src/store/index.ts

```diff
@@ -3,10 +3,54 @@ import Vuex from 'vuex';
 
 Vue.use(Vuex);
 
+type TUser = { userId: number; nickName: string; email: string };
+type TDataSnackbar = {
+  visible: boolean;
+  message: string;
+  color: string;
+  timeout: number;
+};
+
+interface IState {
+  user: TUser;
+  dataSnackbar: TDataSnackbar;
+  isLogged: boolean;
+}
+
+const userDefault: TUser = {
+  userId: 0,
+  nickName: '',
+  email: '',
+};
+
 export default new Vuex.Store({
-  state: {},
+  state: {
+    user: {
+      userId: 0,
+      nickName: '',
+      email: '',
+    },
+    dataSnackbar: {
+      visible: false,
+      message: '',
+      color: 'info',
+      timeout: 5000,
+    },
+    isLogged: false,
+  } as IState,
+
+  mutations: {
+    saveUser(state: IState, user: TUser = userDefault) {
```

Comment:
it makes sense to name mutations in the upper kebab case: SAVE_USER

Reply:
Done